components/api-resource-mgt/org.wso2.carbon.identity.api.resource.mgt/src/main/java/org/wso2/carbon/identity/api/resource/mgt/constant/SQLConstants.java

```diff
@@ -113,6 +123,20 @@ public class SQLConstants {
             " S.DESCRIPTION AS SCOPE_DESCRIPTION" +
             " FROM API_RESOURCE AR LEFT JOIN SCOPE S ON AR.ID = S.API_ID WHERE AR.ID = ? AND (AR.TENANT_ID = ?" +
             " OR AR.TENANT_ID IS NULL)";
+    public static final String GET_API_RESOURCE_BY_ID_FOR_ORGANIZATIONS = "SELECT" +
+            " AR.ID AS API_RESOURCE_ID," +
+            " AR.NAME AS API_RESOURCE_NAME," +
+            " AR.IDENTIFIER AS API_RESOURCE_IDENTIFIER," +
+            " AR.DESCRIPTION AS API_RESOURCE_DESCRIPTION," +
+            " AR.TENANT_ID AS API_RESOURCE_TENANT_ID," +
+            " AR.TYPE AS API_RESOURCE_TYPE," +
+            " AR.REQUIRES_AUTHORIZATION AS REQUIRES_AUTHORIZATION," +
+            " S.ID AS SCOPE_ID," +
+            " S.NAME AS SCOPE_QUALIFIED_NAME," +
+            " S.DISPLAY_NAME AS SCOPE_DISPLAY_NAME," +
+            " S.DESCRIPTION AS SCOPE_DESCRIPTION" +
+            " FROM API_RESOURCE AR LEFT JOIN SCOPE S ON AR.ID = S.API_ID WHERE AR.ID = ? AND (AR.TENANT_ID = ?" +
+            " OR AR.TENANT_ID IS NULL) AND AR.TYPE != 'TENANT' AND AR.TYPE != 'SYSTEM'";
```

Comment:
Why do't we have DB-specific select queries. Does this query works on all of our supported DBs?

Reply:
As of now this was tested against H2, MySQL and MSSQL. For those this is working as expected. Other DB type testings are in progress.